File: cems-SPA/src/app/dashboard/javascript/models/JavascriptBrowserInfo.model.ts
export class JavascriptBrowserInfo {
  public osName: string;
  public osVersion: any;
  public browserName: string;
  public browserVersion: any;
  public browserUserAgent: string;
  public browserLanguage: string;
  public browserIsOnline: boolean;
  public browserPlatform: string;
  public javaEnabled: boolean;
  public dataCookiesEnabled: boolean;
  public dataCookies: string;
  public dataStorage: any;

  public screenScreenW: number;
  public screenScreenH: number;
  public sizeInnerW: number;
  public sizeInnerH: number;
  public screenAvailW: number;
  public screenAvailH: number;
  public scrColorDepth: number;
  public scrPixelDepth: number;

  constructor(javascriptBrowserInfoObj) {
    this.osName = javascriptBrowserInfoObj.osName;
    this.osVersion = javascriptBrowserInfoObj.osVersion;
    this.browserName = javascriptBrowserInfoObj.browserName;
    this.browserVersion = javascriptBrowserInfoObj.browserVersion;
    this.browserUserAgent = javascriptBrowserInfoObj.browserUserAgent;
    this.browserLanguage = javascriptBrowserInfoObj.browserLanguage;
    this.browserIsOnline = javascriptBrowserInfoObj.browserIsOnline;
    this.browserPlatform = javascriptBrowserInfoObj.browserPlatform;
    this.javaEnabled = javascriptBrowserInfoObj.javaEnabled;
    this.dataCookiesEnabled = javascriptBrowserInfoObj.dataCookiesEnabled;
    this.dataCookies = javascriptBrowserInfoObj.dataCookies;
    this.dataStorage = JSON.parse(javascriptBrowserInfoObj.dataStorage);

    this.screenScreenW = javascriptBrowserInfoObj.screenScreenW;
    this.screenScreenH = javascriptBrowserInfoObj.screenScreenH;
    this.sizeInnerW = javascriptBrowserInfoObj.sizeInnerW;
    this.sizeInnerH = javascriptBrowserInfoObj.sizeInnerH;
    this.screenAvailW = javascriptBrowserInfoObj.screenAvailW;
    this.screenAvailH = javascriptBrowserInfoObj.screenAvailH;
    this.scrColorDepth = javascriptBrowserInfoObj.scrColorDepth;
    this.scrPixelDepth = javascriptBrowserInfoObj.scrPixelDepth;
  }

}

File: cems.API/Models/javascript/JavascriptBrowserInfoModel.cs
namespace cems.API.Models.javascript
{
    public class JavascriptBrowserInfoModel
    {
        public string osName { get; set; }
        public string osVersion { get; set; }
        public string browserName { get; set; }
        public string browserVersion { get; set; }
        public string browserUserAgent { get; set; }
        public string browserLanguage { get; set; }
        public bool browserIsOnline { get; set; }
        public string browserPlatform { get; set; }
        public bool javaEnabled { get; set; }
        public bool dataCookiesEnabled { get; set; }
        public string dataCookies { get; set; }
        public string dataStorage { get; set; }

        public int screenScreenW { get; set; }
        public int screenScreenH { get; set; }
        public int sizeInnerW { get; set; }
        public int sizeInnerH { get; set; }
        public int screenAvailW { get; set; }
        public int screenAvailH { get; set; }
        public int scrColorDepth { get; set; }
        public int scrPixelDepth { get; set; }
    }
}